### ISSUE TITLE
$ node utils/update_info.js

### ISSUE THREAD

**Issue by ArtyLisen** [HashLips/hashlips_art_engine#436]
It seems when i try to update info, in only updates the _metadata.json file under build/json. Is this normal?

**Comment by zobtheblob** [comment]
im having the same issues :(

**Comment by largeweb** [comment]
It gives me a message that the name, description, baseUri has all changed but they haven't changed anywhere.

**Comment by largeweb** [comment]
If you know javascript, changing the code is not very hard at all to get it to work. Took me like 15 mins to do exactly what I need.

**Comment by friyays** [comment]
@largeweb can you share what you came up with?

**Comment by largeweb** [comment]
I can't recall exactly, but I believe I made a new file in utils called updateBaseURIOnly.js, copied the updateinfo.js code, and deleted most of it, kept what I thought was sort of the code I needed and just read through it line by line to understand it and after 20 mins I got the right code. I only needed to change the URI.

**Comment by satok111** [comment]
@largeweb could you share your code?  I just can't get it to work.

**Comment by largeweb** [comment]
Sure, here are the two files I made in the utils folder. *You may have to add inside package.json the keyval pair "type": "module", to get the import statements to work. I believe I got an error if I didn't add this keyval pair. Also, read through to make sure the folders are correct, I changed the name of some of my directories, you should be able to easily spot if I used a different directory.

updateBaseURIOnly: (Updates the baseURI based on the metaData.file_url variable)

`import FormData from 'form-data';
import fetch from 'node-fetch';
import path from 'path';
import fs from 'fs';
const basePath = process.cwd();

fs.readdirSync(`${basePath}/build/images`).forEach(file => {
    const fileName = path.parse(`${basePath}/build/json/${file}`).name;
    let rawdata = fs.readFileSync(`${basePath}/build/json/${fileName}.json`);
    let metaData = JSON.parse(rawdata);
    metaData.file_url = `ipfs://INSERT-YOUR-IPFS-GATEWAY-TOKEN-HERE/${fileName}.png`;
    fs.writeFileSync(`${basePath}/build/json/${fileName}.json`, JSON.stringify(metaData, null, 2));
    console.log(`${fileName}.json metadata baseUri should be updated!`);
})

var splitTest = function (str) {
    return str.split('\\').pop().split('/').pop();
}`

bigMetaMakeFile.js: (merges all the metadata files into one big _metadata.json - I only needed this because my big file was not correct)

`const fs = require("fs");
const path = require("path");
const { createCanvas, loadImage } = require("canvas");
const basePath = process.cwd();
const buildDir = `${basePath}/build/jsonEntireFolderTest`;
const console = require("console");
const metadataList = [];

let count = 1;

const writeMetaData = (_data) => {
    fs.writeFileSync(`${buildDir}/_metadata.json`, _data);
};

fs.readdirSync(`${basePath}/build/jsonEntireFolderTest`).forEach(file => {
    if(count != 10001) {
        const fileName = path.parse(`${basePath}/build/json/${file}`).name;
        let rawdata = fs.readFileSync(`${basePath}/build/json/${fileName}.json`);
        let metaData = JSON.parse(rawdata);
        metadataList.push(metaData);
        console.log(`${fileName}.json added to metaDataList`);
        count++;
    }
})

writeMetaData(JSON.stringify(metadataList, null, 2));`

**Comment by satok111** [comment]
ok, thanks

**Comment by patrickmack89** [comment]
> Sure, here are the two files I made in the utils folder. *You may have to add inside package.json the keyval pair "type": "module", to get the import statements to work. I believe I got an error if I didn't add this keyval pair. Also, read through to make sure the folders are correct, I changed the name of some of my directories, you should be able to easily spot if I used a different directory.
> 
> updateBaseURIOnly: (Updates the baseURI based on the metaData.file_url variable)
> 
> `import FormData from 'form-data'; import fetch from 'node-fetch'; import path from 'path'; import fs from 'fs'; const basePath = process.cwd();
> 
> fs.readdirSync(`${basePath}/build/images`).forEach(file => { const fileName = path.parse(`${basePath}/build/json/${file}`).name; let rawdata = fs.readFileSync(`${basePath}/build/json/${fileName}.json`); let metaData = JSON.parse(rawdata); metaData.file_url = `ipfs://INSERT-YOUR-IPFS-GATEWAY-TOKEN-HERE/${fileName}.png`; fs.writeFileSync(`${basePath}/build/json/${fileName}.json`, JSON.stringify(metaData, null, 2)); console.log(`${fileName}.json metadata baseUri should be updated!`); })
> 
> var splitTest = function (str) { return str.split('\').pop().split('/').pop(); }`
> 
> bigMetaMakeFile.js: (merges all the metadata files into one big _metadata.json - I only needed this because my big file was not correct)
> 
> `const fs = require("fs"); const path = require("path"); const { createCanvas, loadImage } = require("canvas"); const basePath = process.cwd(); const buildDir = `${basePath}/build/jsonEntireFolderTest`; const console = require("console"); const metadataList = [];
> 
> let count = 1;
> 
> const writeMetaData = (_data) => { fs.writeFileSync(`${buildDir}/_metadata.json`, _data); };
> 
> fs.readdirSync(`${basePath}/build/jsonEntireFolderTest`).forEach(file => { if(count != 10001) { const fileName = path.parse(`${basePath}/build/json/${file}`).name; let rawdata = fs.readFileSync(`${basePath}/build/json/${fileName}.json`); let metaData = JSON.parse(rawdata); metadataList.push(metaData); console.log(`${fileName}.json added to metaDataList`); count++; } })
> 
> writeMetaData(JSON.stringify(metadataList, null, 2));`

I'm trying to use your code for the big metadata file as mine is messed up also but I'm getting an error.

PS C:\Users\patri\OneDrive\Desktop\hashlips_art_engine-main> node utils/bigMetaMakeFile.js
C:\Users\patri\OneDrive\Desktop\hashlips_art_engine-main\utils\bigMetaMakeFile.js:1
const fs = require("fs"); const path = require("path"); const { createCanvas, loadImage } = require("canvas"); const basePath = process.cwd(); const buildDir = ${basePath}/build/jsonEntireFolderTest`;

             ^

SyntaxError: Unexpected token '{'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    at Module._compile (node:internal/modules/cjs/loader:1067:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47



Any idea why?

**Comment by bolshoytoster** [comment]
@patrickmack89 you need an extra backtick.
change
```js
const buildDir = ${basePath}/build/jsonEntireFolderTest`;
```
to
```js
const buildDir = `${basePath}/build/jsonEntireFolderTest`;
```

**Comment by largeweb** [comment]
Yes it looks like you forgot a backtick, using JSX syntax you need to always use backticks.
